apps/camera/js/app.js

```diff
@@ -363,7 +363,9 @@ App.prototype.onBeforeUnload = function() {
  * @private
  */
 App.prototype.loadL10n = function(done) {
-  this.require(['l10n'], done);
+  this.require(['l10n'], function() {
+    this.require(['l10n_date'], done);
+  });
```

Comment:
Or actually, maybe in config/require.js you can just add l10n_date as a dependency of media_frame and maybe things will just work right. MediaFrame only gets loaded when the user previews photos, so l10n_date shouldn't have to be on the startup path as it presumably is here.